LwM2M client sample cleanup

mike-scott opened the pull request:
- Simplify / cleanup the prj.conf

mike-scott commented:
@therealprof feel free to review

therealprof reviewed (approved):
Fantastic, works a treat!

mike-scott commented on samples/net/lwm2m_client/prj.conf:
```diff
@@ -1,21 +1,14 @@
-CONFIG_SYS_LOG_NET_BUF_LEVEL=2
+CONFIG_SYS_LOG=y
 CONFIG_SYS_LOG_LWM2M_LEVEL=4
-CONFIG_TEST_RANDOM_GENERATOR=y
```
@therealprof Oh interesting!  What hardware is this on?

therealprof replied:
Nucleo-F207

mike-scott replied:
Ok, I'll back this 1 line removal out of the cleanup patch.  I was under the impression that it wasn't being used anymore.  Thank you for reporting.

mike-scott commented:
Re-added the CONFIG_TEST_RANDOM_GENERATOR=y config.

jukkar commented on samples/net/lwm2m_client/src/lwm2m-client.c:
```diff
@@ -344,8 +346,34 @@ static void rd_client_event(struct lwm2m_ctx *client,
+	int ret;
+
+#if defined(CONFIG_NET_IPV6)
+	ret = lwm2m_rd_client_start(&client, CONFIG_NET_APP_PEER_IPV6_ADDR,
```
Please note that the stack might be quite small (CONFIG_NET_MGMT_EVENT_STACK_SIZE, default size is 512 bytes) if event_iface_up() is called from event mgmt thread. This is why I did not call the processing function directly in subsys/net/ip/dhcpv4.c. Have you made sure that lwm2m_rd_client_start() can be called from such a small stack?

mike-scott replied:
I was debugging this issue a few days ago.  And you are quite right.  Under normal circumstances, this works just fine.  But with DTLS enabled there are a few too many nested calls in the networking layer and I end up with a stack crash.  How are other samples doing this?

jukkar replied:
One option is to use system workqueue from event handler as then the stack will be much larger. So search k_delayed_work_submit() from samples/net to see some examples.

mike-scott replied:
Make sense.  I can re-implement here with a delayed work.

mike-scott replied:
@jukkar I worked on this today and adding the delayed_work here in the sample seems.. very fragile.   Instead, I'm going to add the delayed_work to the "start" function in the LwM2M subsys (making sure we're on the main stack no matter how it's called).  Otherwise, issues like this could crop up unexpectedly for users.

My other thought was to add the interface event handling to the subsys as well.  This way less and less code would be needed in the sample to produce reliable results.

jukkar replied:
Ok, makes sense.

mike-scott committed:
samples: net: lwm2m: cleanup prj.conf

- Remove NET_LOG / NET_BUF_LOG settings, instead just enable SYS_LOG
- Simplify by also removing SYS_LOG_SHOW_COLOR, INIT_STACKS and
  NET_STATISTICS.  These can be enabled by user if wanted/needed.

Signed-off-by: Michael Scott <mike@foundries.io>

mike-scott commented:
New version posted:
- As mentioned above, dropped the interface event changes.
- Rebased on today's master branch
